src/modules/media/dto/media.dto.ts

```diff
@@ -1,224 +1,63 @@
-import { InternalServerErrorException } from '@nestjs/common';
 import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
 import { Transform, Type } from 'class-transformer';
-import {
-	IsArray,
-	IsDateString,
-	IsEnum,
-	IsNumber,
-	IsOptional,
-	IsString,
-	ValidateNested,
-} from 'class-validator';
+import { IsArray, IsEnum, IsNumber, IsOptional, IsString, ValidateNested } from 'class-validator';
 import { isArray } from 'lodash';
 
-import { MediaFormat } from '../types';
+import { Operator, SearchFilterField } from '../types';
 
 import { SortDirection } from '~shared/types';
 
-export class RangeQueryDuration {
+export class SearchFilter {
 	@IsString()
-	@IsOptional()
-	@ApiPropertyOptional({
-		type: String,
-		description: 'The minimum duration of the media items',
-		example: '00:15:00',
-	})
-	gte?: string;
-
-	@IsString()
-	@IsOptional()
-	@ApiPropertyOptional({
-		type: String,
-		description: 'The maximum duration of the media items',
-		example: '00:50:00',
-	})
-	lte?: string;
-}
-export class RangeQueryDate {
-	@IsDateString()
-	@IsOptional()
-	@ApiPropertyOptional({
-		type: String,
-		description: 'The minimum date',
-		example: '2020-09-01',
-	})
-	gte?: string;
-
-	@IsDateString()
-	@IsOptional()
-	@ApiPropertyOptional({
-		type: String,
-		description: 'The maximum date',
-		example: '2020-12-31',
-	})
-	lte?: string;
-}
-
-export class AdvancedQuery {
-	@IsString()
-	@IsOptional()
-	@ApiPropertyOptional({
-		type: String,
-		description: 'Filter on items that contain this string',
-	})
-	contains?: string;
-
-	@IsString()
-	@IsOptional()
-	@ApiPropertyOptional({
-		type: String,
-		description: 'Filter on items that do not contain this string',
-	})
-	containsNot?: string;
-
-	@IsString()
-	@IsOptional()
-	@ApiPropertyOptional({
+	@IsEnum(SearchFilterField)
+	@ApiProperty({
 		type: String,
-		description: 'Filter on items that equal this string',
+		description: `The field to filter on. Options are: ${Object.values(SearchFilterField).join(
+			', '
+		)}`,
 	})
-	is?: string;
+	field: SearchFilterField;
 
-	@IsString()
+	@IsArray()
 	@IsOptional()
-	@ApiPropertyOptional({
-		type: String,
-		description: 'Filter on items that do not equal this string',
+	@Transform((input) => {
+		if (!isArray(input.value)) {
+			return [input.value.trim()];
+		}
+		return input.value.map((kw) => kw.trim());
 	})
-	isNot?: string;
-}
-
-export class SearchFilters {
-	@IsString()
-	@IsOptional()
 	@ApiPropertyOptional({
 		type: String,
-		description:
-			'The text to search for in the title, description and abstract of the media items.',
-		default: '',
+		description: `The array of values for the filter. Uses the OR operator. If both multiValue and value are set, value is ignored.`,
 	})
-	query?: string;
+	multiValue?: Array<string>;
```

Comment:
Nice to have: have a single value param, that can either be an array or a single value. But i'm not sure if this will work:
```
@ApiProperty({
   oneOf: [
      { type: 'string' },
      { 
         type: 'array',
         items: {
            type: 'string'
         }
      }
   ]
})
```